src/config.rs

```diff
@@ -694,18 +704,21 @@ impl Cfg {
     }
 
     pub fn list_toolchains(&self) -> Result<Vec<String>> {
-        if utils::is_directory(&self.toolchains_dir) {
-            let mut toolchains: Vec<_> = utils::read_dir("toolchains", &self.toolchains_dir)?
-                .filter_map(io::Result::ok)
-                .filter(|e| e.file_type().map(|f| !f.is_file()).unwrap_or(false))
-                .filter_map(|e| e.file_name().into_string().ok())
-                .collect();
-
-            utils::toolchain_sort(&mut toolchains);
+        let mut toolchains: Vec<_> = self.list_toolchains_iter()?.collect();
+        utils::toolchain_sort(&mut toolchains);
+        Ok(toolchains)
+    }
 
-            Ok(toolchains)
+    fn list_toolchains_iter(&self) -> Result<Box<dyn Iterator<Item = String>>> {
```

Comment:
These days, could that be a `Result<impl Iterator<Item=String>>` ?

Reply:
I don't think so. I tried that, but it didn't work because the `else` branch returns `iter::empty()`, which is a different concrete type.